src/dbnode/storage/database_test.go

```diff
@@ -1032,6 +1033,104 @@ func TestWideQuery(t *testing.T) {
 	assert.Equal(t, exSpans, spanStrs)
 }
 
+func TestReadMismatches(t *testing.T) {
+	ctrl := xtest.NewController(t)
+	defer ctrl.Finish()
+
+	d, mapCh, _ := defaultTestDatabase(t, ctrl, BootstrapNotStarted)
+	defer func() {
+		close(mapCh)
+	}()
+
+	commitlog := d.commitLog
+	d.commitLog = nil
+	ns := dbAddNewMockNamespace(ctrl, d, "testns")
+	nsOptions := namespace.NewOptions()
+	ns.EXPECT().OwnedShards().Return([]databaseShard{}).AnyTimes()
+	ns.EXPECT().Tick(gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+	ns.EXPECT().ShardBootstrapState().Return(ShardBootstrapStates{}).AnyTimes()
+	ns.EXPECT().Options().Return(nsOptions).AnyTimes()
+	require.NoError(t, d.Open())
+
+	ctx := context.NewContext()
+	// create initial span from a mock tracer and get ctx
+	mtr := mocktracer.New()
+	sp := mtr.StartSpan("root")
+	ctx.SetGoContext(opentracing.ContextWithSpan(stdlibctx.Background(), sp))
+
+	var (
+		q = index.Query{
+			Query: idx.NewTermQuery([]byte("foo"), []byte("bar")),
+		}
+		err error
+
+		now      = time.Now()
+		iterOpts = index.IterationOptions{}
+		wideOpts = index.WideQueryOptions{
+			StartInclusive:   now.Truncate(2 * time.Hour),
+			EndExclusive:     now.Truncate(2 * time.Hour).Add(2 * time.Hour),
+			IterationOptions: iterOpts,
+			BatchSize:        1024,
+		}
+	)
+	ctx.SetGoContext(opentracing.ContextWithSpan(stdlibctx.Background(), sp))
+
+	shards := []uint32{1, 2, 3}
+	ns.EXPECT().WideQueryIDs(gomock.Any(), q, gomock.Any(), gomock.Any()).
+		DoAndReturn(func(
+			_ context.Context,
+			_ index.Query,
+			collector chan *ident.IDBatch,
+			opts index.WideQueryOptions,
+		) error {
+			assert.Equal(t, opts.StartInclusive, wideOpts.StartInclusive)
+			assert.Equal(t, opts.EndExclusive, wideOpts.EndExclusive)
+			assert.Equal(t, opts.IterationOptions, wideOpts.IterationOptions)
+			assert.Equal(t, opts.BatchSize, wideOpts.BatchSize)
+			assert.Equal(t, opts.ShardsQueried, shards)
+			go func() {
+				batch := &ident.IDBatch{IDs: []ident.ID{ident.StringID("foo")}}
+				batch.Add(1)
+				collector <- batch
+				close(collector)
+			}()
+			return nil
+		})
+
+	batchReader := wide.NewMockIndexChecksumBlockBatchReader(ctrl)
+	ns.EXPECT().FetchReadMismatches(gomock.Any(), batchReader,
+		ident.StringID("foo"), wideOpts.StartInclusive).
+		Return(wide.EmptyStreamedMismatchBatch, nil)
+	_, err = d.ReadMismatches(ctx, ident.StringID("testns"), q, batchReader, now, shards, iterOpts)
+	require.NoError(t, err)
+
+	ns.EXPECT().WideQueryIDs(gomock.Any(), q, gomock.Any(), gomock.Any()).
+		Return(fmt.Errorf("random err"))
+	_, err = d.ReadMismatches(ctx, ident.StringID("testns"), q, batchReader, now, nil, iterOpts)
```

Comment:
Micro nit (here and above): insert an empty line before `d.ReadMismatches`, will be easier for the eyes to locate where the SuT is being excercised.